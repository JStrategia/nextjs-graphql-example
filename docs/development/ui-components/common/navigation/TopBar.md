# TopBar

Renders a top bar (menu bar) to be used on top of pages.

## Usage

```tsx
function MyComponent() {
  return (
    <>
      <TopBar />
      <span>My content</span>
    </>
  );
}
```
