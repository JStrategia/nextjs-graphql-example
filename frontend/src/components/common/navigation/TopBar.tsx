import React from 'react';
import styled, { css } from 'styled-components';
import { useTranslation } from 'react-i18next';
import Body1 from 'components/common/typography/Body1';
import { Theme } from 'styles';

function TopBar() {
  const { t } = useTranslation();

  // TODO: Add language switcher
  // TODO: Add theme switcher
  // TODO: Add user menu
  return (
    <TopBarContainer>
      <Body1>{t('nextRecipes')}</Body1>
      <SettingsContainer>
        <Body1>Lang</Body1>
        <Body1>Theme</Body1>
        <Body1>User</Body1>
      </SettingsContainer>
    </TopBarContainer>
  );
}

const TopBarContainer = styled.div(
  ({ theme }: { readonly theme: Theme }) => css`
    display: flex;
    align-items: start;
    justify-content: space-between;
    background-color: ${theme.colors.surface};
    padding: ${theme.spacing.m};
  `,
);

const SettingsContainer = styled.div(
  ({ theme }: { readonly theme: Theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: ${theme.spacing.m};
  `,
);

export default TopBar;
