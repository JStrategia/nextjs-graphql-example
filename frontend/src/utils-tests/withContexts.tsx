import React from 'react';
import ThemeProvider from 'contexts/ThemeProvider';

function withContexts(children: React.ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
