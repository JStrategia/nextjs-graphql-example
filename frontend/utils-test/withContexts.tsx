import React, { ReactNode } from 'react';
import ThemeProvider from '../contexts/ThemeProvider';

function withContexts(children: ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
