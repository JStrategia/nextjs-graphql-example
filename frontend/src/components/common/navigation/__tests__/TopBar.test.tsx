import React from 'react';
import { create } from 'react-test-renderer';
import TopBar from 'components/common/navigation/TopBar';
import withContexts from 'utils-tests/withContexts';

describe('TopBar', () => {
  describe('When rendering', () => {
    const render = create(withContexts(<TopBar />)).toJSON();

    test('Then match snapshot', () => {
      expect(render).toMatchSnapshot();
    });
  });
});
