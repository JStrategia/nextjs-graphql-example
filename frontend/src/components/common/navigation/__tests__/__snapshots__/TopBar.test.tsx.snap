// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar When rendering Then match snapshot 1`] = `
<div
  className="sc-gsnTZi dyNsfE"
>
  <span
    className="sc-bczRLJ dqVkrV"
  >
    nextRecipes
  </span>
  <div
    className="sc-dkzDqf iopiUR"
  >
    <span
      className="sc-bczRLJ dqVkrV"
    >
      Lang
    </span>
    <span
      className="sc-bczRLJ dqVkrV"
    >
      Theme
    </span>
    <span
      className="sc-bczRLJ dqVkrV"
    >
      User
    </span>
  </div>
</div>
`;
